include/nlohmann/json.hpp

```diff
@@ -496,8 +496,8 @@ class basic_json
     using object_t = ObjectType<StringType,
           basic_json,
           object_comparator_t,
-          AllocatorType<std::pair<const StringType,
-          basic_json>>>;
+          // Note: this forces object_t::value_type to match std::map's
```

Comment:
I am not sure whether I understood this comment.

Reply:
Please remove the comment and leave that part of the file as is.

Reply:
I added this comment to highlight the restriction imposed by an AllocatorType always specialized for `std::pair<const StringType, basic_json>`. I only managed to make `std::vector`  work through a tough workaround, but you may want to have a more flexible interface in the future.
To be flexible you could e.g. wrap `std::map` like this:
```
template <class Key, class T, class Comparator, template AllocatorTemplate>
class wrapped_map : public std::map<Key, T, Comparator, AllocatorTemplate<const Key, T>>
...
```

Reply:
Anyway I removed the comment to move on with this :-)